Set up automatic building

PoolloverNathan opened the pull request:
This pull request sets up automated compiling and building via Github Actions. The action will create an artifact named "jars" that contains both the distribution jar and the sources jar. To see what the build process looks like, [approve this workflow](https://github.com/astei/lazydfu/actions/runs/4111728291).

This pull request is best suited for rebasing due to only containing a single commit.

Kichura commented on .github/workflows/build.yml:
```diff
@@ -5,8 +5,23 @@ on:
+        with:
+          name: jars
+          path: build/libs
+          if-no-files-found: error
```
Should be "warn" instead of error because otherwise, it would always assume compilation failed.

PoolloverNathan replied:
Fixed, but I used error so that if building jars didn't produce any jars, the build would tell you that something is wrong.

Kichura commented on .github/workflows/build.yml:
```diff
@@ -5,8 +5,23 @@ on:
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+      - name: Install Java 17
+        uses: actions/setup-java@v3.9.0
```
No need to use ".9.0" in setup-java context.

PoolloverNathan replied:
Fixed

Kichura commented on .github/workflows/build.yml:
```diff
@@ -5,8 +5,23 @@ on:
+        uses: actions/setup-java@v3.9.0
+        with:
+          java-version: 17 # Can be more specific if neccesary
+          distribution: adopt
```
"temurin" should go here instead of adopt.

PoolloverNathan replied:
Fixed

Kichura commented on .github/workflows/build.yml:
```diff
@@ -5,8 +5,23 @@ on:
-      - uses: actions/checkout@v2
-      - uses: gradle/wrapper-validation-action@v1
+      - name: Checkout
+        uses: actions/checkout@v2
```
Update this to checkout v3 please.

PoolloverNathan replied:
Done

Kichura reviewed (changes requested):
Looks good but some changes should be considered before hand.

PoolloverNathan committed:
Set up automatic building

This commit adds multiple actions to build.yml that download java 17, build the mod, and upload the jars to Github Actions.

Kichura reviewed (approved):
Looks good now.
